### PR TITLE
[WIP] problem: missing beacon curve support

### PR DESCRIPTION
**Pull request by wesyoung**
solution: add curve support for beacon

dep on:  https://github.com/zeromq/czmq/pull/1757#issuecomment-328579702

(working through all the circle tests...)

caveats:

+ requires --enable-drafts=yes
+ changes beacon version when security enabled
+ starts to implement https://github.com/zeromq/rfc/tree/master/43
+ as things mature- will require a look at zre.xml to change version number... (zproto?) wasn't sure we wanted to mess with this just yet

it turns out- there's a VERSION for beacon packets and a VERSION for ZRE packets. from what i'm understanding, that version has always been the same. but as we added CURVE to the beacons, we iterated the beacon version when security is enabled- but that has zero effect on the ZRE message itself (HELLO- etc).

so it may make sense jus to set everything to v3 before we move to stable? (which may kill interop between v2 and v3 nodes), or something else (break beacon out into an rfc if it isn't already?) or.. other other?

### CODE REVIEW

**Comment by keent** [comment]
@wesyoung Great commit. Thanks much

File: src/zyre.c [hunk 1]
```diff
@@ -592,7 +613,7 @@ zyre_test (bool verbose)
     if (verbose)
         printf ("\n");
 
-    //  @selftest
+    //  @selftestzsys_info ("Using key %s for %s", server_key, endpoint);
```
Comment (bluca) [inline src/zyre.c]: copypasta?

**Comment by bluca** [comment]
The beacon doesn't have a separate RFC, does it? If they are kept separate enough, and there's no inter-dependency, we could bump just one.

File: src/zyre.c (hunk 1 above)
Comment (wesyoung) [reply src/zyre.c]: probably :(

https://github.com/zeromq/zyre/pull/552

**Comment by wesyoung** [comment]
it does not- re-thinking it the VERSION in beacon *does* advertise what version of the actual protocol you're using... they're just ... separate packet structures.. have to mess around with it a bit i guess.

**Comment by wesyoung** [comment]
its prob one of those where you could iterate them separately but they are kind of complementary to each-other too.

**Comment by bluca** [comment]
Ok, makes sense, let's keep it simple

**Pull request merged**